src/test/java/patching/TestDiff.java

```diff
@@ -132,132 +120,39 @@ public void testConvertBack() {
-        result = diff2.transform(Arrays.asList(diff1));
-        Assert.assertEquals(1, result.size());
-        Assert.assertEquals(4, result.get(0).getStartIndex());
-        Assert.assertEquals(2, result.get(0).getLength());
-
-        // the quick ||||||
-        // the k ||||||
-        // the qu ||||||
-        // the ||||||
-        diff1 = new Diff(false, 4, "quic");
-        diff2 = new Diff(false, 6, "ick");
-        result = diff2.transform(Arrays.asList(diff1));
-        Assert.assertEquals(1, result.size());
-        Assert.assertEquals(4, result.get(0).getStartIndex());
-        Assert.assertEquals(1, result.get(0).getLength());
-
-        // the quick brown fox ||||||
-        // the quick fox ||||||
-        // the k brown fox ||||||
-        // the k fox ||||||
-        diff1 = new Diff(false, 10, "brown");
-        diff2 = new Diff(false, 4, "quic");
-        result = diff2.transform(Arrays.asList(diff1));
-        Assert.assertEquals(1, result.size());
-        Assert.assertEquals(4, result.get(0).getStartIndex());
-        Assert.assertEquals(4, result.get(0).getLength());
-
-        // the quick brown fox||||||
-        // the brown fox||||||
-        // the quick brown ||||||
-        // the brown ||||||
-        diff1 = new Diff(false, 4, "quick");
-        diff2 = new Diff(false, 16, "fox");
-        result = diff2.transform(Arrays.asList(diff1));
-        Assert.assertEquals(1, result.size());
-        Assert.assertEquals(11, result.get(0).getStartIndex());
-        Assert.assertEquals(3, result.get(0).getLength());
-
-        // abcdefgh
-        // abpklfgh // Remove "cde", then add "pkl"
-        // abcdgh // Remove "ef"
-        // abpklgh // Should have a net removal of "cdef", add "pkl"
-        diff1 = new Diff(false, 2, "cde");
-        diff2 = new Diff(true, 2, "pkl");
-        diff3 = new Diff(false, 4, "ef");
-        result = diff3.transform(Arrays.asList(diff1, diff2));
-        Assert.assertEquals(1, result.size());
-        Assert.assertEquals(5, result.get(0).getStartIndex());
-        Assert.assertEquals(1, result.get(0).getLength());
+    public void testUndo() {
+        Diff diff = new Diff("0:+4:str1");
+        Diff newDiff = diff.getUndo();
+        Assert.assertEquals("0:-4:str1", newDiff.toString());
+        Diff originalDiff = newDiff.getUndo();
+        Assert.assertEquals(diff.toString(), originalDiff.toString());
+
+        diff = new Diff("1:-4:str2");
+        newDiff = diff.getUndo();
+        Assert.assertEquals("1:+4:str2", newDiff.toString());
+        originalDiff = newDiff.getUndo();
+        Assert.assertEquals(diff.toString(), originalDiff.toString());
     }
 
-
     @Test
     public void testTransform1A() {
         Diff diff1, diff2;
         List<Diff> result;
 
         diff1 = new Diff(true, 2, "str1");
         diff2 = new Diff(true, 4, "str2");
-        result = diff2.transform(Arrays.asList(diff1));
+        result = diff2.transform(Collections.singletonList(diff1));
```

Comment:
Intellij was whining about using Arrays.asList on a single element. I assume that's a style point.